AirCasting/Map/GoogleMapView.swift

```diff
@@ -60,43 +60,69 @@ struct GoogleMapView: UIViewRepresentable {
             let coordinate = point.location
             path.add(coordinate)
         }
-        let polyline = GMSPolyline(path: path)
-        
+        let polyline = context.coordinator.polyline
         let spans = pathPoints.map { point -> GMSStyleSpan in
             let color = colorPolyline(point: point)
             return GMSStyleSpan(style: GMSStrokeStyle.solidColor(color),
                                 segments: 1)
         }
+
+        polyline.path = path
+        polyline.spans = spans
+        polyline.strokeWidth = 10
+        polyline.map = uiView
         
-        let updatedCameraPosition = setStartingPoint(points: pathPoints)
-        DispatchQueue.main.async {
-            uiView.camera = updatedCameraPosition
-        }
-        // Update starting point
-        if !context.coordinator.didSetInitLocation && !pathPoints.isEmpty {
+        // Update camera's starting point
+        if context.coordinator.shouldAutoTrack {
             let updatedCameraPosition = setStartingPoint(points: pathPoints)
             DispatchQueue.main.async {
-                uiView.camera = updatedCameraPosition
+                uiView.moveCamera(cameraUpdate)
             }
-            context.coordinator.didSetInitLocation = true
         }
-        polyline.spans = spans
-        polyline.strokeWidth = 6
-        polyline.map = uiView
+    }
+    
+    var cameraUpdate: GMSCameraUpdate {
+        guard !pathPoints.isEmpty else {
+            let location = tracker.googleLocation.last?.location ?? CLLocationCoordinate2D(latitude: 0, longitude: 0)
+            return GMSCameraUpdate.setTarget(location)
+        }
+        
+        var north = -90.0
+        var south = 90.0
+        var east = -180.0
+        var west = 180.0
+        
+        pathPoints
+            .forEach { point in
+                let latitude = point.location.latitude
+                let longitude = point.location.longitude
+                
+                north = max(north, latitude)
+                south = min(south, latitude)
+                east = max(east, longitude)
+                west = min(west, longitude)
+            }
+        
+        let southWest = CLLocationCoordinate2D(latitude: south, longitude: west)
+        let northEast = CLLocationCoordinate2D(latitude: north, longitude: east)
```

Comment:
as a challenge you can try to do it all using `pathPoints.reduce`

Reply:
Challenge accepted 😎

I know you already approved but could you please take a look if I did it right?